src/app/(routes)/blog/[slug]/_components/article-content.tsx

```diff
@@ -0,0 +1,147 @@
+    ) {
+      return (
+        <IframeContainer>
+          <iframe
+            width="100%"
+            height="100%"
+            src={href}
+            title="YouTube video player"
+            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
+            allowFullScreen
+            style={{
+              position: "absolute",
+              top: 0,
+              left: 0,
+              clipPath: "inset(0% 0% 0% 0% round 16px)",
+            }}
+          ></iframe>
+        </IframeContainer>
+      );
+    }
+    return (
+      <Link
+        target={isExternal(href) ? "_blank" : undefined}
+        className="hover:text-text underline"
+        href={href}
+        type="external"
+      >
+        {children}
+      </Link>
+    );
+  },
+  [BLOCKS.DOCUMENT]: (_, children) => children,
+  [BLOCKS.PARAGRAPH]: (_, children) => (
+    <Text variant="body">
+      <p>{children}</p>
+    </Text>
+  ),
+  [BLOCKS.HEADING_1]: (_, children) => (
+    <Text variant="heading-1" className="py-4">
+      <h1>{children}</h1>
+    </Text>
+  ),
+  [BLOCKS.HEADING_2]: (_, children) => (
+    <Text variant="heading-3" className="py-4">
+      <h2>{children}</h2>
+    </Text>
+  ),
+  [BLOCKS.HEADING_3]: (_, children) => (
+    <Text variant="heading-4">
+      <h3>{children}</h3>
+    </Text>
+  ),
+  [BLOCKS.HEADING_4]: (_, children) => (
+    <Text variant="body">
+      <h4>{children}</h4>
+    </Text>
+  ),
+  [BLOCKS.HEADING_5]: (_, children) => (
+    <Text variant="body">
+      <h5>{children}</h5>
+    </Text>
+  ),
+  [BLOCKS.HEADING_6]: (_, children) => (
+    <Text variant="body">
+      <h6>{children}</h6>
+    </Text>
+  ),
+  [BLOCKS.EMBEDDED_RESOURCE]: (_, children) => <div>{children}</div>,
+  [BLOCKS.UL_LIST]: (_, children) => <ul className="list-disc pl-8">{children}</ul>,
+  [BLOCKS.OL_LIST]: (_, children) => <ol className="list-decimal pl-8">{children}</ol>,
+  [BLOCKS.LIST_ITEM]: (_, children) => <li>{children}</li>,
+  [BLOCKS.QUOTE]: (_, children) => <blockquote>{children}</blockquote>,
+  [BLOCKS.HR]: () => <Divider />,
+  [BLOCKS.TABLE]: (_, children) => (
+    <table>
+      <tbody>{children}</tbody>
+    </table>
+  ),
+  [BLOCKS.TABLE_ROW]: (_, children) => <tr>{children}</tr>,
+  [BLOCKS.TABLE_HEADER_CELL]: (_, children) => <th>{children}</th>,
+  [BLOCKS.TABLE_CELL]: (_, children) => <td>{children}</td>,
+  [BLOCKS.EMBEDDED_ASSET]: (node) => {
+    const data = node.data.target as Asset<"WITHOUT_UNRESOLVABLE_LINKS", string>;
+    const { file, description, title } = data.fields;
+    const mimeGroup = file?.contentType.split("/")[0]; // image / video etc
+    switch (mimeGroup) {
+      case "image":
+        return <ContentfulImage image={data} />;
+      // TODO: test this, make custom component if necessary
+      case "video":
+        return (
+          <video title={title} aria-description={description} src={file?.url}>
```

Comment:
I think the `src` here also needs the protocol, same as in`<ContentfulImage />`.
```ts
const urlWithProtocol = `https:${file?.url}`;
```

Reply:
https://github.com/across-protocol/across.to/pull/56/commits/5f11c09dbff6ea8520f7f4de6946926715136f19